datalad/downloaders/base.py

```diff
@@ -228,42 +264,56 @@ def _enter_credentials(self, url, denied_msg, new=True):
         """
         title = "{msg} access to {url} has failed.".format(
             msg=denied_msg, url=url)
-        if new:
+
+        if new_provider:
             # No credential was known, we need to create an
             # appropriate one
-            if not self.authenticator:
-                raise DownloadError(title +
-                                    " No authenticator is known, cannot set "
-                                    "any credential")
-            credential_type = self.authenticator.DEFAULT_CREDENTIAL_TYPE
-            if not credential_type:
-                raise DownloadError(
-                    title +
-                    " %s does not have a default credential type, "
-                    "cannot authenticate"
-                    % self.authenticator
-                )
-            action_msg = "enter new %s credentials?" % credential_type
-        else:
-            credential_type = None  # just to please pylint
-            action_msg = "enter other credentials in case they were updated?"
-        if ui.yesno(
-                title=title,
-                text="Do you want to %s" % action_msg):
-            if new:
-                # TODO:
-                #  - ask for a type of credential if default not known
-                #  - make it persistent - ask for all the needed fields and save
-                #    in user config directory
+            if not ui.yesno(
+                    title=title,
+                    text="Would you like to setup a new provider "
+                         "configuration to "
+                         "access url?",
+                    default=True
+            ):
+                if not self.authenticator:
+                    raise DownloadError(title +
+                                        " No authenticator is known, cannot "
+                                        "set "
+                                        "any credential")
+                # we could try to just
+                credential_type = self.authenticator.DEFAULT_CREDENTIAL_TYPE
+                if not credential_type:
+                    raise DownloadError(
+                        title +
+                        " %s does not have a default credential type, "
+                        "cannot authenticate"
```

Comment:
Why put "any credential" on a new line?

Reply:
what `"any credential"`? ah -- way above! will fix and compress here a bit. thanks